libsplinter/src/admin/shared.rs

```diff
@@ -514,24 +518,29 @@ impl AdminServiceShared {
     pub fn submit(&mut self, payload: CircuitManagementPayload) -> Result<(), ServiceError> {
         debug!("Payload submitted: {:?}", payload);
 
-        match self.verify_signature(&payload) {
-            Ok(_) => (),
-            Err(ServiceError::UnableToHandleMessage(_)) => (),
-            Err(err) => return Err(err),
-        };
-
         let header =
             protobuf::parse_from_bytes::<CircuitManagementPayload_Header>(payload.get_header())?;
+        self.validate_circuit_management_payload(&payload, &header)
+            .map_err(|err| ServiceError::UnableToHandleMessage(Box::new(err)))?;
+
+        self.verify_signature(&payload)?;
```

Comment:
Either split this out into its own commit or add a message to the commit.

Reply:
Split handling the unknown / unset actions into a separate commit and added a description of the signature verification to that existing commit.